proxy/src/config.rs

```diff
@@ -380,3 +415,65 @@ fn parse<T, Parse>(strings: &Strings, name: &str, parse: Parse) -> Result<Option
         None => Ok(None),
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    fn test_units<F: Fn(u64) -> Duration>(units: &[&str], to_duration: F) {
+        for v in &[1, 23, 456_789] {
```

Comment:
We should include 0 here too. We should consider whether we support the value 0 at all for any of these values, and, if so, whether we require a unit for 0.